rubicon/objc/objc.py

```diff
@@ -1314,6 +1326,65 @@ def __setattr__(self, name, value):
                 super(ObjCInstance, type(self)).__setattr__(self, name, value)
 
 
+class ObjCListInstance(ObjCInstance):
+    def __getitem__(self, item):
+        if item >= self.count:
+            raise IndexError('list index out of range')
+        return self.objectAtIndex(item)
+
+    def __setitem__(self, item, value):
+        try:
+            self.insertObject_atIndex_(value, item)
+        except AttributeError:
+            name = self.objc_class.name
+            raise TypeError("'%s' object does not support item assignment" % name)
+
+    def __len__(self):
+        return self.count
+
+    def __iter__(self):
+        for i in range(self.count):
+            yield self.objectAtIndex(i)
+
+    def __contains__(self, item):
+        return self.containsObject_(item)
+
+
+class ObjCDictInstance(ObjCInstance):
+    def __getitem__(self, item):
+        v = self.objectForKey_(item)
+        if v is None:
+            raise KeyError(repr(item))
```

Comment:
Minor issue: the default Python behavior is to pass the missing key directly to `KeyError`, and not its `repr`:

```python
>>> import sys
>>> d = {}
>>> d[42]
Traceback (most recent call last):
  File "<pyshell#1>", line 1, in <module>
    d[42]
KeyError: 42
>>> sys.last_value.args[0]
42
```